services/QuillLMS/app/models/previous_year_school_datum.rb

```diff
@@ -0,0 +1,26 @@
+class PreviousYearSchoolDatum
+  include VitallySchoolStatsHelper
+
+  def initialize(school, year)
+    @year = year
+    @school = school
+  end
+
+  def calculate_and_save_data
```

Comment:
`and` in a method name is usually a good clue to break a method/class/object into two things.

I would keep the PreviousYearSchoolDatum model, but have a separate object for the `VitallyPreviousSchoolYearCache`.  For example, abstracted out the ClassroomsCache for GoogleIntegration [here](https://github.com/empirical-org/Empirical-Core/blob/64b6acaabe3c46747e7a993b5b3e44ede4e0f774/services/QuillLMS/app/services/google_integration/users/classrooms_cache.rb) That way there's one place for CRUD actions on the cache.